File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    var answer = -1;
    let array=s.split('')
    let length=s.length;
    // 길이 홀수면 0 반환
    if(s.length % 2 ===1)
        return 0;
    
    // 문자열 줄이는 과정
    let stack = [];
    for (let i = 0; i < s.length; i++) {
        stack.push(s[i]);
        if (stack.length > 1 && stack[stack.length - 1] === stack[stack.length - 2]) {
            stack.pop();
            stack.pop();
        }
    }
    
    if(stack.length>0)
        return 0;
    else
        return 1;
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        int answer = -1;
        Deque<Character> stack = new ArrayDeque<Character>();
        
        for(int i=0;i<s.length();i++){
            if(stack.isEmpty()){
                stack.push(s.charAt(i));
            }else{
                Character prev = stack.peek();
                if(prev==s.charAt(i)){
                    stack.pop();
                }else{
                    stack.push(s.charAt(i));
                }
            }
        }

        if(stack.isEmpty()){
            return 1;
        }else{
            return 0;
        }
    }
}